### DISCUSSION TITLE
Deployment Error Using Ganache (msg.value)

### DISCUSSION THREAD

**Discussion by mrghubba** [ConsenSys-archive/truffle#5450] [General Q&A]
Hello. I'm having an issue with 'msg.value' within my 0x token swap API code. Here is my code:
https://pastebin.com/RfCH8gMk
I'm using Ganache within Remix IDE.
When I deploy sending 1 e18 Wei, the transaction is reverted:
https://paste.pics/I5EML
When I deploy sending 0 Wei and I fill in the parameters in the 'Pay' function......
https://paste.pics/I4LA1
I get the 'Empty Call Data Without Swap' Error:
https://paste.pics/I5EO1
Is there a way around this?

**Comment by kevinbluer** [comment]
Hey @mrghubba! As per the other [discussion topic](https://github.com/orgs/trufflesuite/discussions/5447)...are you referring to this project: https://github.com/0xProject/0x-api-starter-guide-code? If so I'd probably recommend doing it all locally given the [README](https://github.com/0xProject/0x-api-starter-guide-code/blob/master/README.md) requires a number of dependencies, configuration, etc.

If not, a repo with both the contract code (as provided in the pastebin) along with all the other config, scripts, etc would be helpful to help troubleshoot too if that's something you can share?
